src/backend/storage/buffer/bufmgr.c

```diff
@@ -995,6 +996,8 @@ ReadBuffer_start(SMgrRelation smgr, char relpersistence, ForkNumber forkNum,
 	/* Make sure we will have room to remember the buffer pin */
 	ResourceOwnerEnlargeBuffers(CurrentResourceOwner);
 
+	isExtend = (blockNum == P_NEW);
```

Comment:
ReadBuffer_start() isn't called for file extensions, so this doesn't really make sense. Note the `Assert(blockNum != P_NEW)` above.